service/controller/resource/tcnp/create.go

```diff
@@ -275,12 +275,17 @@ func newAutoScalingGroup(ctx context.Context, cr infrastructurev1alpha2.AWSMachi
 		Cluster: template.ParamsMainAutoScalingGroupCluster{
 			ID: key.ClusterID(&cr),
 		},
-		DesiredCapacity:       minDesiredNodes,
-		MaxBatchSize:          workerCountRatio(minDesiredNodes, 0.3),
-		MaxSize:               key.MachineDeploymentScalingMax(cr),
-		MinInstancesInService: workerCountRatio(minDesiredNodes, 0.7),
-		MinSize:               key.MachineDeploymentScalingMin(cr),
-		Subnets:               subnets,
+		DesiredCapacity:                     minDesiredNodes,
+		MaxBatchSize:                        workerCountRatio(minDesiredNodes, 0.3),
+		MaxSize:                             key.MachineDeploymentScalingMax(cr),
+		MinInstancesInService:               workerCountRatio(minDesiredNodes, 0.7),
+		MinSize:                             key.MachineDeploymentScalingMin(cr),
+		Subnets:                             subnets,
+		OnDemandPercentageAboveBaseCapacity: 0,
+		//If spot instance then 0, else 100
+		OnDemandBaseCapacity:    0,
```

Comment:
We need the flag for spot instances here to either set this to 0% or 100%. 100% means all instances will be on-demand. 0% means all instances will be spot instances.

Reply:
:+1: here is a nice table explaining the different values https://docs.aws.amazon.com/autoscaling/ec2/userguide/asg-purchase-options.html